File: Backend/TrackrAPI/TrackrAPI/Services/Catalogo/PaisService.cs
using TrackrAPI.Repositorys;
using TrackrAPI.Models;
using System.Collections.Generic;
using TrackrAPI.Repositorys.Catalogo;
using TrackrAPI.Dtos.Catalogo;

namespace TrackrAPI.Services.Catalogo
{
    public class PaisService
    {
        private IPaisRepository paisRepository;
        private PaisValidatorService paisValidatorService;

        public PaisService(IPaisRepository paisRepository, PaisValidatorService paisValidatorService)
        {
            this.paisRepository = paisRepository;
            this.paisValidatorService = paisValidatorService;
        }

        public IEnumerable<PaisDto> ConsultarTodosParaSelector()
        {
            return paisRepository.ConsultarTodosParaSelector();
        }

        public PaisDto ConsultarDto(string clave)
        {
            return paisRepository.ConsultarDto(clave);
        }

        public PaisDto ConsultarDto(int idPais)
        {
            return paisRepository.ConsultarDto(idPais);
        }

        public IEnumerable<PaisGridDto> ConsultarGeneral()
        {
            return paisRepository.ConsultarGeneral();
        }

        public void Agregar(Pais pais)
        {
            paisValidatorService.ValidarAgregar(pais);
            paisRepository.Agregar(pais);
        }

        public void Editar(Pais pais)
        {
            paisValidatorService.ValidarEditar(pais);
            paisRepository.Editar(pais);
        }

        public void Eliminar(int idPais)
        {
            Pais pais = paisRepository.Consultar(idPais);
            paisValidatorService.ValidarEliminar(idPais);
            paisRepository.Eliminar(pais);
        }

    }
}

File: Frontend/trackr-administrador/src/app/shared/http/catalogo/pais.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Pais } from '@models/catalogo/pais';

@Injectable()
export class PaisService {
  private dataUrl = 'pais/';

  constructor(public http: HttpClient) {}

  consultarTodosParaSelector() {
    return this.http.get<Pais[]>(this.dataUrl + 'consultarTodosParaSelector');
  }

  consultar(idPais: number) {
    return this.http.get<Pais>(this.dataUrl + `consultar/${idPais}`);
  }

  consultarPorClave(clave: string) {
    return this.http.get<Pais>(this.dataUrl + `consultarPorClave/${clave}`);
  }

  consultarGeneral() {
    return this.http.get<Pais[]>(this.dataUrl + 'consultarGeneral');
  }

  agregar(pais: Pais) {
    return this.http.post<void>(this.dataUrl + 'agregar', pais);
  }

  editar(pais: Pais) {
    return this.http.put<void>(this.dataUrl + 'editar', pais);
  }

  eliminar(idPais: number) {
    return this.http.delete<void>(this.dataUrl + `eliminar/${idPais}`);
  }
}